java/com/walmartlabs/logback/RiemannAppender.java

```diff
@@ -140,28 +150,49 @@ protected synchronized void append(E event) {
 
     if (isMinimumLevel(logEvent)) {
       EventDSL rEvent = createRiemannEvent(logEvent);
+
+      if (debug) {
+        printError("%s.append: sending riemann event: %s", className, rEvent);
+      }
+
       try {
-        try {
-          if (debug) {
-            printError("%s.append: sending riemann event: %s", className, rEvent);
-          }
-          rEvent.send();
-          if (debug) {
-            printError("%s.append(logEvent): sent to riemann %s:%s", className, riemannHostname, riemannPort);
-          }
-        } catch (Exception ex) {
-          if (debug) {
-            printError("%s: Error sending event %s", this, ex);
-            ex.printStackTrace(System.err);
-          }
 
-          riemannClient.reconnect();
-          rEvent.send();
-        }
-      } catch (Exception ex) {
-        // do nothing
+        int attempt = 0;
```

Comment:
Looking at this again, I just realized that the behavior is actually no different than the previous version. Because the new code is substantially larger than the old, I'd assumed the retry logic was new. 

I have to admit: I don't think this is easier to read. We still have nested `try` blocks, it's just that now we also have a loop interposed between them. I'd prefer to see this written as before, maybe with a method extracted to eliminate the duplication around checking for the acknowledgement.